src/ensembl/production/metadata/service.py

```diff
@@ -364,6 +364,78 @@ def get_genome_by_name(metadata_db, ensembl_name, site_name, release_version):
         return create_genome()
 
 
+def populate_datasets_info(data):
+    ds_info = {
+        'dataset_uuid': data['dataset_uuid'],
+        'dataset_name': data['dataset_name'],
+        'dataset_version': data['dataset_version'],
+        'dataset_label': data['dataset_label'],
+        'version': data['version']
+    }
+    return ds_info
+
+
+def get_datasets_by_uuid(metadata_db, genome_uuid, release_version):
+    # This is sqlalchemy's metadata, not Ensembl's!
+    md = db.MetaData()
+    session = Session(metadata_db, future=True)
+
+    # Reflect existing tables, letting sqlalchemy load linked tables where possible.
+    genome = db.Table('genome', md, autoload_with=metadata_db)
+    dataset = db.Table('dataset', md, autoload_with=metadata_db)
+    genome_dataset = db.Table('genome_dataset', md, autoload_with=metadata_db)
+    dataset_type = db.Table('dataset_type', md, autoload_with=metadata_db)
+    ensembl_release = db.Table('ensembl_release', md, autoload_with=metadata_db)
+
+    genome_select = db.select(
+        genome.c.genome_uuid,
+        dataset.c.dataset_uuid,
+        # The label here should be identical to what's in proto file
+        dataset_type.c.name.label('data_set_type'),
+        dataset.c.name.label('dataset_name'),
+        dataset.c.version.label('dataset_version'),
+        dataset.c.label.label('dataset_label'),
+        ensembl_release.c.version
+        ).select_from(
+            genome
+        ).join(genome_dataset).join(dataset) \
+        .join(dataset_type).join(ensembl_release) \
+        .where(genome.c.genome_uuid == genome_uuid) \
+        .where(genome_dataset.c.is_current == 1) \
+        .distinct()
+
+    if release_version > 0:
+        genome_select = genome_select.filter_by(version=release_version)
```

Comment:
Might be interesting to do a `<=`, because I am still not sure whether we'll replicate version attachment for everything every time we do a new release.